src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -0,0 +1,51 @@
+package com.kodilla.ecommercee.controller;
+
+import com.kodilla.ecommercee.domain.Cart;
+import com.kodilla.ecommercee.domain.CartDto;
+import com.kodilla.ecommercee.mapper.CartMapper;
+import com.kodilla.ecommercee.service.CartService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Objects;
+
+@RestController
+@RequestMapping("v1/cart")
+public class CartController {
+    @Autowired
+    private CartService cartService;
+
+    @Autowired
+    private CartMapper cartMapper;
+
+    @GetMapping
+    public Cart createEmptyCart() {
+        return cartService.getNewCart();
+    }
+
+    @GetMapping(value="getById")
```

Comment:
Chyba metody w REST nie miały być nazywane? Mielismy się opierac na typach żądań

Reply:
Tak, ale jedną z metod @GetMapping musiałem nazwać. W sytuacji, kiedy więcej niż raz zastosuję metodę @GetMapping i obie metody nazywają się tak samo, czyli po prostu nie mają swojej nazwy - program wywala błędy

Reply:
No bo masz druga,  która tworzy encje a to nie jest zgodne z REST.

Reply:
I wiem, ze program wywala błędy jak sa dwa @GetMapping bo sam na to Wam zwrocilem uwagę.  Była tez w tej konwersacji od razu uwaga od innego mentora ze GET  nie powinien tworzyć niczego w bazie... a tutaj nadal tak  jest

Reply:
Ok, to w takim razie createEmptyCart powinien zwracać obiekt klasy CartDto, zamiast Cart?

Reply:
Nie, nie powinien. Niech zwraca to co zwraca ale musi uzywac poprawnej metody HTRP

Reply:
ja jednak bym sugerował korzystać z ustaleń zespołu:
GET /v1/product - wszystkie produkty
GET /v1/product/5 albo /v1/product?id=5 - product po ID
**POST /v1/product - tworzenie nowego produktu**
DELETE /v1/product/ - usuwanie produktu o id 5

więc użyłbym POST zamiast HEAD.